tracking-migration/src/test/java/io/customer/tracking/migration/testutils/extensions/JSONExtensions.kt

```diff
@@ -0,0 +1,74 @@
+package io.customer.tracking.migration.testutils.extensions
+
+import io.customer.sdk.data.model.CustomAttributes
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeInstanceOf
+import org.amshove.kluent.shouldNotBeNull
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * Asserts that the JSONObject is empty.
+ */
+fun JSONObject.shouldBeEmpty() {
+    this.length() shouldBeEqualTo 0
+}
+
+/**
+ * Asserts that JSONObject has the same key-value pairs as the expected CustomAttributes.
+ */
+infix fun JSONObject.shouldMatchTo(expected: CustomAttributes) {
+    this.length() shouldBeEqualTo expected.size
+
+    for (key in keys()) {
+        val actualValue = this.get(key)
+
+        when (val expectedValue = expected[key]) {
+            is Map<*, *> -> actualValue.shouldBeInstanceOf<JSONObject>() shouldMatchTo expectedValue.shouldBeInstanceOf()
+            is List<*> -> actualValue.shouldBeInstanceOf<JSONArray>() shouldMatchTo expectedValue.shouldBeInstanceOf()
+            else -> actualValue shouldBeEqualTo expectedValue
+        }
+    }
+}
+
+/**
+ * Asserts that JSONArray has the same values as the expected List.
+ */
+infix fun JSONArray.shouldMatchTo(expected: List<Any>) {
+    this.length() shouldBeEqualTo expected.size
+
+    for (index in 0 until this.length()) {
+        val actualValue = this[index]
+
+        when (val expectedValue = expected[index]) {
+            is Map<*, *> -> actualValue.shouldBeInstanceOf<JSONObject>() shouldMatchTo expectedValue.shouldBeInstanceOf()
+            is List<*> -> actualValue.shouldBeInstanceOf<JSONArray>() shouldMatchTo expectedValue.shouldBeInstanceOf()
+            else -> actualValue shouldBeEqualTo expectedValue
+        }
+    }
+}
```

Comment:
```suggestion
infix fun JSONArray.shouldMatchTo(expected: List<Any>) {
    length() shouldBeEqualTo expected.size

    (0 until length()).forEach { index ->
        val actualValue = this[index]
        when (val expectedValue = expected[index]) {
            is Map<*, *> -> actualValue.shouldBeInstanceOf<JSONObject>().shouldMatchTo(expectedValue)
            is List<*> -> actualValue.shouldBeInstanceOf<JSONArray>().shouldMatchTo(expectedValue)
            else -> actualValue shouldBeEqualTo expectedValue
        }
    }
}
```

totally optional 😄 just more idiomatic.